File: past201912/past201912_d.cs
// https://atcoder.jp/contests/past201912-open/submissions/9638205

using System;

namespace CSharp
{
    class Program
    {
        public static void Main()
        {
            var N = int.Parse(Console.ReadLine());

            var t = new int[N];
            var d = 0;
            for (int i = 0; i < N; i++)
            {
                var input = int.Parse(Console.ReadLine());
                if (++t[input - 1] == 2)
                {
                    d = input;
                }
            }

            if (d > 0)
            {
                Console.WriteLine($"{d} {Array.IndexOf(t, 0) + 1}");
            }
            else
            {
                Console.WriteLine("Correct");
            }
        }
    }
}

File: past201912/past201912_d.fs
// https://atcoder.jp/contests/past201912-open/submissions/13505374

let N = stdin.ReadLine() |> int
let A = Array.zeroCreate<int> N

let mutable d = 0

for i in [ 0..N - 1 ] do
    let a = stdin.ReadLine() |> int
    A.[a - 1] <- A.[a - 1] + 1
    if A.[a - 1] = 2 then d <- a

if d > 0 then d.ToString() + " " + ((A |> Array.findIndex ((=) 0)) + 1).ToString()
else "Correct"
|> stdout.WriteLine
